src/components/Menu/MenuItem.tsx

```diff
@@ -118,32 +152,65 @@ class MenuItem extends UIComponent<Extendable<MenuItemProps>, MenuItemState> {
     wrapper: { as: 'li' },
   }
 
-  state = {
-    isFromKeyboard: false,
+  static autoControlledProps = ['submenuOpen']
+
+  private outsideClickSubscription = EventStack.noSubscription
+
+  private submenuRef = React.createRef<HTMLElement>()
+  private itemRef = React.createRef<HTMLElement>()
+
+  public componentDidMount() {
+    this.updateOutsideClickSubscription()
+  }
+
+  public componentDidUpdate() {
+    this.updateOutsideClickSubscription()
+  }
+
+  public componentWillUnmount() {
+    this.outsideClickSubscription.unsubscribe()
   }
 
-  renderComponent({ ElementType, classes, accessibility, rest }) {
-    const { children, content, icon, wrapper } = this.props
+  renderComponent({ ElementType, classes, accessibility, rest, styles }) {
+    const { children, content, icon, wrapper, menu, primary, secondary, active } = this.props
+
+    const { submenuOpen } = this.state
 
     const menuItemInner = childrenExist(children) ? (
       children
     ) : (
-      <ElementType
-        className={classes.root}
-        onClick={this.handleClick}
-        onBlur={this.handleBlur}
-        onFocus={this.handleFocus}
-        {...accessibility.attributes.anchor}
-        {...accessibility.keyHandlers.anchor}
-        {...rest}
-      >
-        {icon &&
-          Icon.create(this.props.icon, {
-            defaultProps: { xSpacing: !!content ? 'after' : 'none' },
-          })}
-        {content}
-      </ElementType>
+      <Ref innerRef={this.itemRef}>
+        <ElementType
+          className={classes.root}
+          onBlur={this.handleBlur}
+          onFocus={this.handleFocus}
+          {...accessibility.attributes.anchor}
+          {...rest}
+          {...!wrapper && { onClick: this.handleClick }}
+        >
+          {icon &&
+            Icon.create(this.props.icon, {
+              defaultProps: { xSpacing: !!content ? 'after' : 'none' },
+            })}
+          {content}
+        </ElementType>
+      </Ref>
     )
+    const maybeSubmenu =
+      menu && active && submenuOpen ? (
+        <Ref innerRef={this.submenuRef}>
+          {Menu.create(menu, {
+            defaultProps: {
+              accessibility: submenuBehavior,
+              vertical: true,
```

Comment:
is there any possibility that it could be horizontal one? I mean, I might imagine it (horizontal menu and horizontal submenu) - this was the case for Word's toolbar, for example

Reply:
According to the spec: https://github.com/stardust-ui/react/issues/542 the submenus should always be vertical. @jurokapsiar opinion on this?

Reply:
as a proof for my point:

![image](https://user-images.githubusercontent.com/9024564/50227500-27481280-03a6-11e9-8bed-6a37f9a2a812.png)

Reply:
Will be added in a follow up PR.